Missing `const` in doc

In the example in the doc there is this function:
```rust
impl Flag {
    pub fn new(short: char, name: &'static str) -> Self {
        Flag { short, name }
    }
}
```
But it fail compiling with this error: `error[E0015]: cannot call non-const fn 'Flag::new' in statics`. The example file works fine because it uses `const fn new`

--- a/README.md
+++ b/README.md
@@ -45,7 +45,7 @@ pub struct Flag {
 }
 
 impl Flag {
-    pub fn new(short: char, name: &'static str) -> Self {
+    pub const fn new(short: char, name: &'static str) -> Self {
         Flag { short, name }
     }
 }

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -40,7 +40,7 @@
 //! }
 //!
 //! impl Flag {
-//!     pub fn new(short: char, name: &'static str) -> Self {
+//!     pub const fn new(short: char, name: &'static str) -> Self {
 //!         Flag { short, name }
 //!     }
 //! }

